internal/applicationsnapshot/input.go

```diff
@@ -189,59 +240,39 @@ func expandImageIndex(ctx context.Context, snap *app.SnapshotSpec) {
 
 	client := oci.NewClient(ctx)
 	// For an image index, remove the original component and replace it with an expanded component with all its image manifests
-	var components []app.SnapshotComponent
 	// Do not raise an error if the image is inaccessible, it will be handled as a violation when evaluated against the policy
 	// This is to retain the original behavior of the `ec validate` command.
-	var allErrors error = nil
-	for _, component := range snap.Components {
-		// Assume the image is not an image index or it isn't accessible
-		components = append(components, component)
-		ref, err := name.ParseReference(component.ContainerImage)
-		if err != nil {
-			allErrors = errors.Join(allErrors, fmt.Errorf("unable to parse container image %s: %w", component.ContainerImage, err))
-			continue
-		}
 
-		desc, err := client.Head(ref)
-		if err != nil {
-			allErrors = errors.Join(allErrors, fmt.Errorf("unable to fetch descriptior for container image %s: %w", ref, err))
-			continue
-		}
+	componentChan := make(chan app.SnapshotComponent, len(snap.Components))
```

Comment:
This will be too high. I tried these changes with 59 components and quay started to report `429 Too Many Requests`.

I think this should match the value from the `--workers` parameter, default is 5.

With the 59 component snapshot, without your changes, it takes 2m52.847s to expand all the images. With your changes (but capping at 5 concurrency) this takes `0m8.583s`. That's pretty darn good.

Reply:
Or maybe it should be a different value. I can see how we may want a high number for validating images, but a lower number for expanding them. For example, 50 is fine for validating, but it's too high for quay.io.

Consider maybe just hard-coding to something like 5 for now.

Reply:
Ah, Good point. I'll update.

Reply:
Tried this with the `errgroup` package. https://github.com/enterprise-contract/ec-cli/pull/2128/commits/367518b7ce946d3bd6ce79258cc37282714d5511

Reply:
I want to try and test this and see if the limit works.